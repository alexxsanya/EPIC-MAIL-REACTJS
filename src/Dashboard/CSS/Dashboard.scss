
@import '../../common/CSS/assets/variables';
@import '../../common/CSS/assets/fonts';

.dashboard {
  margin-top: 20px;
  min-width: 90%;

  .side-section {
    width: 10%;
  }

  .nav-link {
    background: $white;
    border-bottom: 1px solid #dfdfdf;

    &:hover {
      background: $dark;
      color: #555;
    }
  }

  .content-section {
    background: inherit;
  }

  .section-title {
    border-bottom: 1px solid $white;
    color: $primary-blue;
    font-family: $lato-font;
    font-size: 24px;
    font-weight: 100;
    margin: 0 0 10px 0;
    text-align: left;
    width: 100%;
  }

  .list-group-item{
    border-radius: 0 !important;
  }
}
