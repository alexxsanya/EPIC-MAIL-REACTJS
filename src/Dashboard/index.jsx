import React from 'react';
import {
  Container,
  Row,
  Col,

} from 'react-bootstrap';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Loader from 'react-loader';
import store from '../Mainstore/Store';
import './CSS/Dashboard.scss';
import StatusLabel from '../common/components/StatusLabel';
import SideMenu from './components/SideMenu';
import InboxMessages from './components/InboxMessages';

const Dashboard = () => (

  <Provider store={store}>
    <Container className="dashboard">
      <Row>
        <Col sm={2} className="side-section">
          <SideMenu />
        </Col>
        <Col sm={10} className="content-section">
          <Router>
            <Switch>
              <Route path="/dashboard/inbox" exact component={InboxMessages} />
              <Route path="/dashboard/send" exact component={StatusLabel} />
            </Switch>
          </Router>
        </Col>
      </Row>
    </Container>
  </Provider>
);
export default Dashboard;
