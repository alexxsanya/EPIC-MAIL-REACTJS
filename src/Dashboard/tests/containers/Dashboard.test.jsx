/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dashboard from '../../index';

describe('Dashboard Component', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Dashboard />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
