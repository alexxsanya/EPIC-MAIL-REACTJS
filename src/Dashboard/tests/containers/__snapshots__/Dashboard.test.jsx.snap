// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should match the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Container
    className="dashboard"
    fluid={false}
  >
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="side-section"
        sm={2}
      >
        <SideMenu />
      </Col>
      <Col
        className="content-section"
        sm={10}
      >
        <BrowserRouter>
          <Switch>
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(InboxMessages)",
                  "type": [Function],
                }
              }
              exact={true}
              path="/dashboard/inbox"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(SentMessages)",
                  "type": [Function],
                }
              }
              exact={true}
              path="/dashboard/sent"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/dashboard/send"
            />
          </Switch>
        </BrowserRouter>
      </Col>
    </ForwardRef>
  </Container>
</Provider>
`;
